cpp/include/raft/core/detail/macros.hpp

```diff
@@ -22,6 +22,12 @@
 #endif
 #endif
 
+#if defined(_RAFT_HAS_CUDA)
+#define CUDA_ONLY_CONDITION(condition) condition
+#else
+#define CUDA_ONLY_CONDITION(condition) true
+#endif
```

Comment:
Do you mind future-proofing that with:
```c++
#if defined(_RAFT_HAS_CUDA)
#define CUDA_CONDITION_ELSE_TRUE(condition) condition
#define CUDA_CONDITION_ELSE_FALSE(condition) condition
#else
#define CUDA_CONDITION_ELSE_TRUE(condition) true
#define CUDA_CONDITION_ELSE_FALSE(condition) false
#endif
```

and using `CUDA_CONDITION_ELSE_TRUE`?